### PR TITLE
Remove microk8s from sunbeam

### PR DESCRIPTION
**Pull request by MylesJP**

### CODE REVIEW

**Review by gboutry — APPROVED** [review]
This is looking good, just need a rebase, you can take these comments into consideration then

File: .github/workflows/test-snap.yml [hunk 1]
```diff
@@ -52,13 +52,10 @@ on:
           - single
           - multi
       k8s-provider:
-        description: Select k8s provider
+        description: Specify k8s provider
```
Comment (hemanthnakkina) [inline .github/workflows/test-snap.yml:55]: This job is created to test k8s and microk8s. With only one provider now, we can remove this job.

File: sunbeam-python/sunbeam/core/k8s.py [hunk 2]
```diff
@@ -24,17 +24,14 @@
...
 from sunbeam.core.common import SunbeamException
 
 LOG = logging.getLogger(__name__)
 
-# --- Microk8s specific
-MICROK8S_KUBECONFIG_KEY = "Microk8sConfig"
-MICROK8S_DEFAULT_STORAGECLASS = "microk8s-hostpath"
-
 # --- K8s specific
+METALLB_ANNOTATION = "metallb.universe.tf/loadBalancerIPs"
```
Comment (hemanthnakkina) [inline sunbeam-python/sunbeam/core/k8s.py:34]: This constant is not required as it is replaced with METALLB_IP_NOTATION

**Review by hemanthnakkina — CHANGES_REQUESTED** [review]

File: .github/workflows/test-snap.yml (hunk 1 above)
Comment (hemanthnakkina) [reply .github/workflows/test-snap.yml:55]: Lets keep this job and remove k8s-provider flag as input.

**Review by hemanthnakkina — APPROVED** [review]

**Pull request merged**